src/main/java/com/couchbase/lite/router/URLConnection.java

```diff
@@ -47,12 +45,8 @@
 
     public URLConnection(URL url) {
         super(url);
-        responseInputStream = new PipedInputStream();
-        try {
-            responseOutputStream = new PipedOutputStream((PipedInputStream)responseInputStream);
-        } catch (IOException e) {
-            Log.e(Log.TAG_ROUTER, "Exception creating piped output stream", e);
-        }
+        responseOutputStream = new BufferOutputStream();
+        responseInputStream = new BufferInputStream((BufferOutputStream)responseOutputStream);
```

Comment:
The BufferedOuputStream from the java.io package just adds a buffer of fixed size to any other output stream. This means it will write data into its internal buffer and write the contents of the buffer to the underlying stream when its buffer is full, the flush method is called or the stream gets closed. If writing to the underlying stream blocks, this will still result in a deadlock.